cameo/strain_design/heuristic/evolutionary/objective_functions.py

```diff
@@ -165,36 +172,48 @@ class biomass_product_coupled_yield(YieldFunction):
     doi:10.1186/1471-2105-6-308
     """
 
-    def __init__(self, biomass, product, substrate, *args, **kwargs):
+    def __init__(self, biomass, product, substrate, carbon_yield=False, *args, **kwargs):
         super(biomass_product_coupled_yield, self).__init__(product, substrate, *args, **kwargs)
         if isinstance(biomass, Reaction):
             biomass = biomass.id
         self.biomass = biomass
+        self.carbon_yield = carbon_yield
 
     def __call__(self, model, solution, targets):
-        try:
-            biomass_flux = round(solution.fluxes[self.biomass], config.ndecimals)
+        biomass_flux = round(solution.fluxes[self.biomass], config.ndecimals)
+        if self.carbon_yield:
+            product_flux = round(solution.fluxes[self.product], config.ndecimals) * self.n_c_product
+            substrate_flux = sum(round(abs(solution.fluxes[s]), config.ndecimals) * self.n_c_substrates.get(s, 1)
```

Comment:
Taking the absolute value can in theory give some weird results. If one substrate is suddenly produced for some reason it will still count toward the uptake flux.